src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -67,12 +67,18 @@ class Crystal::Doc::Generator
     end
 
     if filename
-      body = doc(program_type, File.read(filename))
+      raw_body = File.read(filename)
+      body = doc(program_type, raw_body)
     else
+      raw_body = ""
       body = ""
     end
 
     File.write "#{@dir}/index.html", MainTemplate.new(body, types, repository_name)
+
+    main_index = Main.new(raw_body, Type.new(self, @program), repository_name)
+    File.write "#{@dir}/index.json", main_index
+    File.write "#{@dir}/index.jsonp", main_index.to_jsonp
```

Comment:
I have not encountered any other issues with doc pages loaded from local file system and I don't think loading the search index should break that. But I guess we should discuss this in the context of  a subsequent implementation of a builtin search-feature, so for now I'll remove the JSONP file from this PR.